### ISSUE TITLE
Parameter value type "scale" for recipe chrome-pdf not compatible?

### ISSUE THREAD

**Issue by kuakman** [jsreport/jsreport-chrome-pdf#13]
I started running into this issue, whenever I pass the parameter option scale to 1, 1.0 or "1.0" to the recipe 'chrome-pdf' through jsreport.

Here the list of options how it's being received by jsreport:

```
{ scale: '1.0',
  format: 'Letter',
  displayHeaderFooter: true,
  headerTemplate: '<div></div>',
  footerTemplate: '<div class="page-information"\n\t\t\tstyle="font-family: Helvetica, Arial, sans-serif;font-size:7pt!important;\n\t \t\tcolor:grey!important;padding:0;margin-left:.15in;">\n\t\t\tPage&nbsp;<span class="pageNumber"></span>&nbsp;of&nbsp;<span class="totalPages"></span>\n\t\t</div>',
  marginTop: '.25in',
  marginBottom: '.5in',
  marginLeft: '.25in',
  marginRight: '.25in' }
```

This is the error output I get in the jsreport console:

`2018-06-25T16:30:58.779Z - error: Error when processing render request Protocol error (Page.printToPDF): Invalid parameters scale: double value expected Error: Protocol error (Page.printToPDF): Invalid parameters scale: double value expected
    at Promise (jsreports/node_modules/puppeteer/lib/Connection.js:200:56)
    at new Promise (<anonymous>)
    at CDPSession.send jsreports/node_modules/puppeteer/lib/Connection.js:199:12)
    at Page.pdf (jsreports/node_modules/puppeteer/lib/Page.js:798:39)
    at runWithTimeout (jsreports/node_modules/jsreport-chrome-pdf/lib/chrome.js:169:36)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)`

Looking at the source code inside chrome.js on this commit - https://github.com/jsreport/jsreport-chrome-pdf/commit/b92b9ec3d87dc31ae156d05ee800dfd3ce81dc6f#diff-a7c88e4bf351928799b9036ef131dc57R73

The scale value type is causing problems in my case.

Replacing the EMD.String type with EDM.Int32  (which was the original type set previous to that commit), fixes the issue.
Seems like the function Page.printToPdf doesn't like the value type on the scale param to be passed as String for some reason. I haven't dug too deep into the code, but this is symptom I'm getting.

Operative System: Mac OS Sierra 10.12.6
NodeJs: 8.9.1

Thanks!

**Comment by bjrmatos** [comment]
hi! i can confirm the bug that you found. however `Edm.Int32` probably is not the best type for this option because the value can contain decimals. changing to `Edm.Double` or `Edm.Decimal` should be right type, however i will discuss this change internally (because i have one concern about changing the type) first before releasing a patch release. thanks for the bug report.

**Comment by kuakman** [comment]
@bjrmatos Sure no worries! I noticed that bug when trying to pass a different scale value.

**Comment by bjrmatos** [comment]
just to keep the issue updated.. we are doing some changes and soon we will fix this along with other perf improvements

**Comment by kuakman** [comment]
@bjrmatos what I'd suggest is to add a battery of integration tests against all possible parameters, so if something fails, that will block the release. Essentially, if it detects incompatibility value types between puppeteer (```launch()``` method), the instance of the page (```const page = await browser.newPage(); Page.pdf(options);```), and the options parser by jsreport-chrome-pdf.

I think there are a few in already here: https://github.com/jsreport/jsreport-chrome-pdf/blob/master/test/chromeTest.js

Those integration tests will provide solid releases.

Anyways, Thanks for such an amazing tool!

**Comment by bjrmatos** [comment]
fixed in version `1.1.4`. thanks for issue report, about the tests for puppeteer options, we will try to add them in time, for now we can live without it for a while
